### PR TITLE
feat(esl-utils): extend object:compare utils with `isSimilar` method

### PR DESCRIPTION
**Pull request by ala-n**
Relates to: #878

### CODE REVIEW

File: src/modules/esl-utils/misc/object/compare.ts [hunk 1]
```diff
@@ -12,3 +12,12 @@ export function deepCompare(obj1: any, obj2: any): boolean {
   }
   return false;
 }
```
Comment (ala-n) [inline src/modules/esl-utils/misc/object/compare.ts:14]: Proposal: rename `deepCompare` rename to `isEqual`
(non breaking change with current alias)

Feel free to share your reaction for that proposal with emojy or comment

**Review by NastaLeo — APPROVED** [review]

**Review by nattallius — APPROVED** [review]

**Comment by ala-n** [comment]
@NastaLeo , @nattallius, sorry but there is a couple of changes appended to this feature.
Please note that the array checkering strategy reworked.

File: src/modules/esl-utils/misc/object/test/compare.test.ts [hunk 2]
```diff
@@ -1,39 +1,75 @@
...
-    [{a: {a: 1, b: 1}}, {a: {b: 1}}],
-    [{a: {b: 1}}, {a: {a: 1, b: 1}}],
-    [[], [1]],
-    [[1], [1, 2]],
-    [[2, 1], [1, 2]]
-  ])('%p should not be equal to %p', (a: any, b: any) => expect(deepCompare(a, b)).toBe(false));
+import {isEqual, isSimilar} from '../compare';
+
+describe('misc/object: compare', () => {
+  describe('deepCompare', () => {
```
Comment (NastaLeo) [inline src/modules/esl-utils/misc/object/test/compare.test.ts:4]: describe('isEqual', () => {

File: src/modules/esl-utils/misc/object/compare.ts [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+    return !keys1.some((key) => !isEqual(obj1[key], obj2[key]));
   }
   return false;
 }
+
+/** @deprecated */
+export const deepCompare = isEqual;
+
+/**
+ * Checks if all keys presented in the `mask` equal to the `obj` keys
```
Comment (NastaLeo) [inline src/modules/esl-utils/misc/object/compare.ts:20]: Checks if all keys presented in the `mask` are equal to the `obj` keys

**Review by yadamskaya — APPROVED** [review]

**Review by nattallius — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Pull request merged**